app/models/notification.rb

```diff
@@ -228,16 +229,22 @@ def self.recent_report(user, count = nil)
     notifications = notifications.to_a
 
     if notifications.present?
-
-      ids = DB.query_single(<<~SQL, limit: count.to_i)
+      builder = DB.build(<<~SQL)
          SELECT n.id FROM notifications n
-         WHERE
-           n.high_priority = TRUE AND
-           n.user_id = #{user.id.to_i} AND
-           NOT read
+         /*where*/
         ORDER BY n.id ASC
-        LIMIT :limit
+        /*limit*/
+      SQL
+
+      builder.where(<<~SQL, user_id: user.id)
+        n.high_priority = TRUE AND
+        n.user_id = :user_id AND
+        NOT read
       SQL
+      builder.where("notification_type IN (:types)", types: types) if types.present?
```

Comment:
If the user has configured likes notification to be disabled, should the likes types be excluded here?

Reply:
Hmm not sure about this one. If someone calls `Notification.recent_report(someuser, 10, [Notification.types[:liked]])` and `someuser` has likes notifications disabled, what should the method return? Empty list because `someuser` has likes notifications disabled or their (old) like notifications?

Practically though it doesn't matter because this query only looks at high priority notifications and we never mark like notifications as high priority. But that may change in the future, we never know...

Reply:
:ok_hand: Thank you for clarifying. I wasn't quite sure if we had to because but thought it'll be good to ask.